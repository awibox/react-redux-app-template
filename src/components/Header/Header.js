import React from 'react';
import { Link } from 'react-router-dom';
import { projectName } from 'config';
// Components
import NavBar from 'components/NavBar/NavBar';
// Styles
import styles from './Header.scss';

const Header = () => (
  <header className={styles.header}>
    <Link className={styles.logoBlock} to="/">
        <span className={styles.logo}>
            <i className="fa fa-gears"></i>
        </span>
      <span className={styles.name}>{projectName}</span>
    </Link>
    <NavBar/>
  </header>);

export default Header;
