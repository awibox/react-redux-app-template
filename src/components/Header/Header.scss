@import '../../styles/variables';

.header {
  background: $headerBackground;
  width: 100%;
  min-width: $widthScreenMobileSmall;
  border-bottom: 1px solid $headerBorderColor;
  display: flex;
  flex-flow: row;
  box-sizing: border-box;
  justify-content: space-between;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  height: $headerHeight;
}

.logoBlock {
  padding: 10px 0;
  margin-left: $headerIndent;
  display: flex;
  flex-flow: row;
  text-decoration: none;
}

.logo {
  width: 32px;
  height: 32px;
  font-size: 32px;
  color: $headerTextColor;
}

.name {
  font-size: 22px;
  line-height: 32px;
  height: 32px;
  font-weight: 100;
  margin-left: $headerIndent;
  color: $headerTextColor;
}
