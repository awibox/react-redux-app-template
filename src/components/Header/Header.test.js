import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('HomeInfo', () => {
  it('should render correctly', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
