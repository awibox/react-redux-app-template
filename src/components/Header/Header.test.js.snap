// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<header
  className="header"
>
  <Link
    className="logoBlock"
    to="/"
  >
    <span
      className="logo"
    >
      <i
        className="fa fa-gears"
      />
    </span>
    <span
      className="name"
    >
      React Redux App
    </span>
  </Link>
  <NavBar />
</header>
`;
